From 1c4dae066b3b98708d9ce6f78aeea6b8b16dd34f Mon Sep 17 00:00:00 2001
Subject: [PATCH] Android JNI and Vulkan Header Fixes

This patch contains the simple, non-conflicting changes for the Android build.
The more complex changes are handled directly in the workflow via sed.

---
 src/android/app/src/main/jni/native.cpp      | 5 ++---
 src/video_core/vulkan_common/vulkan_device.h | 10 ++++++++++
 2 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/src/android/app/src/main/jni/native.cpp b/src/android/app/src/main/jni/native.cpp
index 6dac1ef84..6fb25d8e8 100644
--- a/src/android/app/src/main/jni/native.cpp
+++ b/src/android/app/src/main/jni/native.cpp
@@ -136,9 +136,8 @@ void EmulationSession::InitializeGpuDriver(const std::string& hook_lib_dir,
     }
 
     // Try to load the system driver.
-    if (!handle) {
-        handle = adrenotools_open_libvulkan(RTLD_NOW, featureFlags, nullptr, hook_lib_dir.c_str(),
-                                            nullptr, nullptr, file_redirect_dir_, nullptr);
+    if (!handle) {
+        handle = dlopen("libvulkan.so", RTLD_NOW);
     }
 
     m_vulkan_library = std::make_shared<Common::DynamicLibrary>(handle);
diff --git a/src/video_core/vulkan_common/vulkan_device.h b/src/video_core/vulkan_common/vulkan_device.h
index 27b80b210c..cb13f28523 100644
--- a/src/video_core/vulkan_common/vulkan_device.h
+++ b/src/video_core/vulkan_common/vulkan_device.h
@@ -377,6 +377,10 @@ public:
 
     /// Returns true if shader int64 is supported.
     bool IsShaderInt64Supported() const {
+        const auto driver = GetDriverID();
+        if (driver == VK_DRIVER_ID_QUALCOMM_PROPRIETARY) {
+            return false;
+        }
         return features.features.shaderInt64;
     }
 
@@ -591,6 +595,10 @@ public:
 
     /// Returns true if the device supports VK_KHR_shader_atomic_int64.
     bool IsExtShaderAtomicInt64Supported() const {
+        const auto driver = GetDriverID();
+        if (driver == VK_DRIVER_ID_QUALCOMM_PROPRIETARY) {
+            return false;
+        }
         return extensions.shader_atomic_int64;
     } 
